### PR TITLE
fix #197 : feat(backend): Added mail verification

### LINKED ISSUE #197: [Feature Request]: Implement the Email verification feature
### Is your feature request related to a problem? Please describe.

Currently there is no way to verify if a user is signing up with their official email address. We need to figure this out

### Describe the solution you'd like

A simple solution would be to send the users a confirmation mail on the email address that they are using to signup.
We can generate a verification token that would be sent via the confirmation mail. This token can be in the form of a link etc. When the user clicks on the link then only we will allow them to create an account.

We will divide this issue in two parts, in the first part we will deal with setting a SMPT server and sending a mail when the user tries to login.
In the second part we will aim to generate the verification token and verify the users when they click on it.

### Describe alternatives you've considered

You will find a preexisting SMTP config, we will start building upon it 

### Developer Help

https://www.youtube.com/watch?v=wB1qOExDsYY
Here's a tutorial that explains the task we are going to perform.

### PR DESCRIPTION
**Pull request by Om-Thorat**
## Overview

1. This PR fixes or fixes part of #197 
2. This PR does the following: Adds mail verification to ensure only the person owning the mail can create an account with the same.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct

https://github.com/bsoc-bitbyte/GetIt/assets/76207818/b2166b62-1dd5-4b93-bf69-2881067c1aa0

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

### CODE REVIEW

**Comment by Om-Thorat** [comment]
Updated the video to reflect recent changes.

**Comment by arminpatel** [comment]
@Om-Thorat let's include some unit tests, for reference you can take a look at the existing tests

**Comment by Om-Thorat** [comment]
> @Om-Thorat let's include some unit tests (for backend part), for reference you can take a look at the existing tests

Edited a test to ensure account is inactive initially, added tests for account verification with a malformed and a valid token.

File: backend/accounts/views.py [hunk 1]
```diff
@@ -30,7 +59,16 @@ def put(self, request):
             return Response(serializer.data)
         return Response(serializer.errors)
 
+def get_account(email) :
+    return get_object_or_404(Account, email=email)
 
+def activate(request, uidb64, token):
```
Comment (arminpatel) [inline backend/accounts/views.py:65]: A more verbose name? maybe `activateAccountView`

File: backend/core/settings/base.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 
 DEBUG = os.environ.get('DEBUG') != 'False'
 
+PRODUCTION_URL = 'https://getit.iiitdmj.ac.in'
```
Comment (arminpatel) [inline backend/core/settings/base.py:16]: imo a env variable is a better option
also what do you mean by production url? let's change it to client_url or frontend_url for now

**Review by arminpatel — CHANGES_REQUESTED** [review]
great work @Om-Thorat
Some minor changes to address

File: backend/accounts/views.py [hunk 3]
```diff
@@ -30,7 +59,16 @@ def put(self, request):
             return Response(serializer.data)
         return Response(serializer.errors)
 
+def get_account(email) :
+    return get_object_or_404(Account, email=email)
 
+def activateAccountView(request, uidb64, token):
```
Comment (arminpatel) [inline backend/accounts/views.py:65]: let's check for request type and only allow a `POST` request

File: backend/core/settings/base.py (hunk 2 above)
Comment (arminpatel) [reply backend/core/settings/base.py]: also we should change the variable name

Comment (Om-Thorat) [reply backend/core/settings/base.py]: oh right got it

File: backend/accounts/tests.py [hunk 4]
```diff
@@ -72,6 +79,36 @@ def testCreateAccountView_invalidOrDuplicateEmail_returnsBadRequest(new_email):
...
+    client = APIClient()
+    account = Account.objects.create(**account_data)
+    token = account_activation_token.make_token(account)
+    uid = urlsafe_base64_encode(force_bytes(account.email))
+
+    # Act
+    response = client.post(f'/api/accounts/activate/{uid}/malformed{token}')
+    print(response.content,response)
+    # Assert
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
```
Comment (arminpatel) [inline backend/accounts/tests.py:94]: also assert that the acc is inactive

**Review by arminpatel — CHANGES_REQUESTED** [review]

File: backend/accounts/tests.py (hunk 4 above)
Comment (Om-Thorat) [reply backend/accounts/tests.py:94]: makes sense, sorry for the oversight.

**Review by arminpatel — APPROVED** [review]

**Review by Kushdapush — APPROVED** [review]

**Pull request merged**